mesonbuild/backend/ninjabackend.py

```diff
@@ -1784,8 +1784,16 @@ def generate_link(self, target, outfile, outname, obj_list, linker, extra_args=[
         commands += self.get_cross_stdlib_link_args(target, linker)
         commands += linker.get_linker_always_args()
         if not isinstance(target, build.StaticLibrary):
-            commands += compilers.get_base_link_args(self.environment.coredata.base_options,
-                                                     linker)
+            options = self.environment.coredata.base_options
+            old_b_lundef = None
+            if (isinstance(target, build.SharedLibrary) and
```

Comment:
We should allow this for both executables and shared libraries, so we can just remove this check. Sometimes even an executable can have a symbol made available at runtime somehow.